mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,301 @@
+                    mrk_idx.append(i_num)
+                if i_mrk != 0 and i_mrk == prev_mrk:
+                    tmp_dur += 1
+                if i_mrk == 0 and i_mrk != prev_mrk:
+                    duration.append((tmp_dur + 1) * (1.0 / srate))
+                    tmp_dur = 0
+                prev_mrk = i_mrk
+            onset = [i_mrk * (1.0 / srate) for i_mrk in mrk_idx]
+            description = [float(i_mrk)for i_mrk in mrk_data[mrk_idx]]
+            annot = Annotations(onset, duration, description)
+            self.set_annotations(annot)
+
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a segment of data from a file.
+
+        Boxy file organises data in two ways, parsed or un-parsed.
+        Regardless of type, output has (n_montages x n_sources x n_detectors
+        + n_marker_channels) rows, and (n_timepoints x n_blocks) columns.
+        """
+        source_num = self._raw_extras[fi]['source_num']
+        detect_num = self._raw_extras[fi]['detect_num']
+        start_line = self._raw_extras[fi]['start_line']
+        end_line = self._raw_extras[fi]['end_line']
+        filetype = self._raw_extras[fi]['filetype']
+        datatype = self._raw_extras[fi]['datatype']
+        boxy_file = self._raw_extras[fi]['file']
+
+        # Possible detector names.
+        detectors = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K',
+                     'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V',
+                     'W', 'X', 'Y', 'Z']
+
+        # Load our optical data.
+        boxy_data = list()
+
+        # Loop through our data.
+        with open(boxy_file, 'r') as data_file:
+            for line_num, i_line in enumerate(data_file, 1):
+                if line_num == (start_line - 1):
+
+                    # Grab column names.
+                    col_names = np.asarray(re.findall(r'\w+\-\w+|\w+\-\d+|\w+',
+                                                      i_line.rsplit(' ')[0]))
+                if (line_num > start_line and line_num <= end_line):
+
+                    # Grab actual data.
+                    boxy_data.append(i_line.rsplit(' '))
+
+        # Get number of sources.
+        sources = np.arange(1, source_num + 1, 1)
+
+        # Grab the individual data points for each column.
+        boxy_data = [re.findall(r'[-+]?\d*\.?\d+', i_row[0])
+                     for i_row in boxy_data]
+
+        # Make variable to store our data as an array
+        # rather than list of strings.
+        boxy_length = len(col_names)
+        boxy_array = np.full((len(boxy_data), boxy_length), np.nan)
+        for ii, i_data in enumerate(boxy_data):
+
+            # Need to make sure our rows are the same length.
+            # This is done by padding the shorter ones.
+            padding = boxy_length - len(i_data)
+            boxy_array[ii] = np.pad(np.asarray(i_data, dtype=float),
+                                    (0, padding), mode='empty')
```

Comment:
We have to accommodate old NumPy (back to 1.13.3 currently) and this causes an error:
```
mne/io/boxy/boxy.py:240: in _read_segment_file
    (0, padding), mode='empty')
../../../miniconda/envs/test/lib/python3.6/site-packages/numpy/lib/arraypad.py:1325: in pad
    for kw in allowedkwargs[mode]:
E   KeyError: 'empty'
```